packages/remix-server-runtime/serialize.ts

```diff
@@ -49,16 +88,45 @@ type DeferValue<T> =
 
 type Pretty<T> = { [K in keyof T]: T[K] };
 
-type Loader<T> = () => Promise<
-  | TypedResponse<T> // returned responses
-  | TypedResponse<never> // thrown responses
->;
+type Loader<T> = () => Promise<TypedResponse<T>>;
```

Comment:
@pcattori and I worked through some examples and we don't need to represent the never types since they don't show up in the signature if you throw conditionally.